rocks-lib/src/project/mod.rs

```diff
@@ -121,6 +139,221 @@ impl Project {
     pub fn tree(&self, lua_version: LuaVersion) -> io::Result<Tree> {
         Tree::new(self.root.join(".rocks"), lua_version)
     }
+
+    pub async fn add(
+        &mut self,
+        dependencies: DependencyType,
+        package_db: &RemotePackageDB,
+    ) -> Result<(), ProjectEditError> {
+        let mut rocks =
+            toml_edit::DocumentMut::from_str(&tokio::fs::read_to_string(self.rocks_path()).await?)?;
+
+        if !rocks.contains_table("dependencies") {
+            let mut table = toml_edit::table().into_table().unwrap();
+            table.set_implicit(true);
+
+            rocks["dependencies"] = toml_edit::Item::Table(table);
+        }
+        if !rocks.contains_table("build_dependencies") {
```

Comment:
question: will this add an empty table if we don't specify any build dependencies to add?

Reply:
Nope, the tables are marked as `implicit` to prevent this :)